src/V8JsiRuntime.cpp

```diff
@@ -1,1621 +1,1624 @@
-    }
-  }
-}
-
-void V8Runtime::ReportException(v8::TryCatch *try_catch) {
-  _ISOLATE_CONTEXT_ENTER
-  v8::Local<v8::Message> message = try_catch->Message();
-  if (message.IsEmpty()) {
-    // V8 didn't provide any extra information about this error; just
-    // throw the exception.
-    v8::String::Utf8Value exception(isolate, try_catch->Exception());
-    throw jsi::JSError(*this, ToCString(exception));
-  } else {
-    std::stringstream sstr;
-
-    // Print (filename):(line number): (message) - this would differ from what JSI expects (it wants the plain
-    // stacktrace)
-    /*v8::String::Utf8Value filename(
-        isolate, message->GetScriptOrigin().ResourceName());
-    v8::Local<v8::Context> context(isolate->GetCurrentContext());
-    const char *filename_string = ToCString(filename);
-    int linenum = message->GetLineNumber(context).FromJust();
-    sstr << filename_string << ":" << linenum << ": " << exception_string
-         << std::endl;
-
-    // Print line of source code
-    v8::String::Utf8Value sourceline(
-        isolate, message->GetSourceLine(context).ToLocalChecked());
-    const char *sourceline_string = ToCString(sourceline);
-    sstr << sourceline_string << std::endl;
-
-    // Print wavy underline (GetUnderline is deprecated).
-    int start = message->GetStartColumn(context).FromJust();
-    for (int i = 0; i < start; i++) {
-      sstr << " ";
-    }
-    int end = message->GetEndColumn(context).FromJust();
-    for (int i = start; i < end; i++) {
-      sstr << "^";
-    }
-    sstr << std::endl;*/
-
-    v8::Local<v8::Value> stack_trace_string;
-    if (try_catch->StackTrace(context_.Get(isolate)).ToLocal(&stack_trace_string) && stack_trace_string->IsString() &&
-        v8::Local<v8::String>::Cast(stack_trace_string)->Length() > 0) {
-      v8::String::Utf8Value stack_trace(isolate, stack_trace_string);
-      const char *stack_trace_string2 = ToCString(stack_trace);
-      sstr << stack_trace_string2 << std::endl;
-    }
-
-    v8::String::Utf8Value ex_message(isolate, message->Get());
-    std::string ex_messages = ToCString(ex_message);
-    if (ex_messages.rfind("Uncaught Error:", 0) == 0) {
-      // V8 adds an "Uncaught Error: " before any message string any time we read it, this strips it out.
-      ex_messages.erase(0, 16);
-    }
-
-    TRACEV8RUNTIME_CRITICAL(
-        "Exception",
-        TraceLoggingString(ex_messages.c_str(), "ex_messages"),
-        TraceLoggingString(sstr.str().c_str(), "sstr"));
-
-    // V8 doesn't actually capture the current callstack (as we're outside of scope when this gets called)
-    // See also https://v8.dev/docs/stack-trace-api
-    if (ex_messages.find("Maximum call stack size exceeded") == std::string::npos) {
-      auto err = jsi::JSError(*this, ex_messages);
-      err.value().getObject(*this).setProperty(
-          *this, "stack", facebook::jsi::String::createFromUtf8(*this, sstr.str()));
-      err.setStack(sstr.str());

```

Comment:
@tudorms this method was removed from `JSError` without any apparent replacement.

Is it safe to assume we can just leave it out?
Maybe the stack is implicitly set now?

Reply:
I think we should just add that in; I added it a while back manually but we never backported to Facebok. It's needed because it's the only way to preserve the callstack in V8 (I guess for Hermes the message includes the stack)

Reply:
Done.